### ISSUE TITLE
Transform test data

### ISSUE THREAD

**Issue by RocaVincent** [Warvito/neurocombat_sklearn#2]
Hi,

If I understood correctly, we pass our target variables as argument of the `fit` method (`discrete_covariates` or `continuous_covariates`). But it seems that we also have to give this variables for the `transform` method. But it's not correct to give variables to `transform` that we want to predict, no ?

**Comment by molly00ecla** [comment]
The combat was a harmonization method, instead of a model. The input and output of the method were both data, so the discrete_covariates or continuous_covariates should also be given.

**Comment by rafaloz** [comment]
Following this question up, although I understand that combat is a method and not a model, the method uses the parameters learned in training and uses in tests. So in this way, I guess there wouldn't be data leakage. However, if you are using the method to predict any of your covariates you will need the covariate for the test transformation, and in that case, my guess is that you are leaking. Is this correct?

**Comment by sahahn** [comment]
> Following this question up, although I understand that combat is a method and not a model, the method uses the parameters learned in training and uses in tests. So in this way, I guess there wouldn't be data leakage. However, if you are using the method to predict any of your covariates you will need the covariate for the test transformation, and in that case, my guess is that you are leaking. Is this correct?

I just found this implementation today, and have only looked at it briefly, but I believe your understanding of the situation is correct. The way they have the code setup is that calling transform on new subjects would be fine (as long as they are from seen sites). That said, as you mentioned, if one of your covariates was also the variable you are predicting, then you would want to be careful as you should not pass this transformer object the real target value, or as you noted, you will introduce data leakage. 

So I actually also recently wrote a sklearn compliant version of Combat (https://github.com/sahahn/neuroCombat/blob/master/neuroCombat/combat.py), and the way I handled this issue was with a include_y attribute, which if set to True, would include your target during training / fit_transform, but then during transform on new subjects will essentially just use the mean of the true target value as a feature. So using this code, the way to replicate that would be just when calling transform on new subjects you would pass the mean of the target variable on the training set as the value (instead of the true target value which you are not supposed to know)

**Comment by rafaloz** [comment]
I checked the implementation and the functions _make_design_matrix and _standardize_across_features have a boolean argument called fitting. This argument is only True if these functions are called during the fit and the argument is False when the functions are called during transform.

 When is True the functions calculate the design matrix and the parameters beta_hat, grand_mean and var_pool to perform the standardization. When the argument is false these parameters are recovered from the fit step. So even though the method requires the input of the covariates when transforming the test set, this set is not used for the final transformation. The input is required to pass the data checks at the beginning of the transformation.

I think this is what @molly00ecla referred to in his message

Thank you!
